api/listener_server_test.go

```diff
@@ -103,3 +103,56 @@ func TestListenerServer(t *testing.T) {
 		return true
 	}, time.Second*2, time.Millisecond*100)
 }
+
+func TestDeleteActiveListener(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	srv, err := api.NewServer(ctx)
+	require.NoError(t, err)
+
+	rec, err := srv.Upsert(ctx, &pb.Record{
```

Comment:
Could we add a second record to confirm deleting non-connected listener deletion still works?

Reply:
added